osu.Game/Screens/SelectV2/BeatmapLeaderboardWedge.cs

```diff
@@ -0,0 +1,390 @@
+                                    },
+                                }
+                            },
+                        },
+                    },
+                    placeholderContainer = new Container<Placeholder>
+                    {
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        RelativeSizeAxes = Axes.Both,
+                    },
+                    loading = new LoadingLayer(),
+                }
+            };
+
+            ((IBindable<LeaderboardScores?>)fetchedScores).BindTo(leaderboardManager.Scores);
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            Scope.BindValueChanged(_ => refetchScores());
+            FilterBySelectedMods.BindValueChanged(_ => refetchScores());
+            beatmap.BindValueChanged(_ => refetchScores());
+            ruleset.BindValueChanged(_ => refetchScores());
+            mods.BindValueChanged(_ => refetchScoresFromMods());
+
+            refetchScores();
+        }
+
+        protected override void PopIn()
+        {
+            this.FadeIn(300, Easing.OutQuint);
+        }
+
+        protected override void PopOut()
+        {
+            this.FadeOut(300, Easing.OutQuint);
+        }
+
+        private void refetchScoresFromMods()
+        {
+            if (FilterBySelectedMods.Value)
+                refetchScores();
+        }
+
+        private void refetchScores()
+        {
+            SetScores(Array.Empty<ScoreInfo>(), null);
+            SetState(LeaderboardState.Retrieving);
+
+            if (beatmap.IsDefault)
+            {
+                SetState(LeaderboardState.NoneSelected);
+                return;
+            }
+
+            var fetchBeatmapInfo = beatmap.Value.BeatmapInfo;
+            var fetchRuleset = ruleset.Value ?? fetchBeatmapInfo.Ruleset;
+
+            if (!api.IsLoggedIn && isOnlineScope)
+            {
+                SetState(LeaderboardState.NotLoggedIn);
+                return;
+            }
+
+            if (!fetchRuleset.IsLegacyRuleset())
+            {
+                SetState(LeaderboardState.RulesetUnavailable);
+                return;
+            }
+
+            if ((fetchBeatmapInfo.OnlineID <= 0 || fetchBeatmapInfo.Status <= BeatmapOnlineStatus.Pending) && isOnlineScope)
+            {
+                SetState(LeaderboardState.BeatmapUnavailable);
+                return;
+            }
+
+            if (Scope.Value.RequiresSupporter(FilterBySelectedMods.Value) && !api.LocalUser.Value.IsSupporter)
+            {
+                SetState(LeaderboardState.NotSupporter);
+                return;
+            }
+
+            if (Scope.Value == BeatmapLeaderboardScope.Team && api.LocalUser.Value.Team == null)
+            {
+                SetState(LeaderboardState.NoTeam);
+                return;
+            }
```

Comment:
All of this should be gone and `LeaderboardManager` should be relied on to produce these error conditions instead. See https://github.com/ppy/osu/pull/32863

Reply:
I'm going to PR a minor refactor to the base class before we copy it here, because I rewrote some of the logic in frenzi's PR sequence around this code without realising it still exists in the old implementation.

Reply:
Revisit this after https://github.com/ppy/osu/pull/32913

Reply:
I've updated the logic @bdach a second set of eyes on it would be appreciated.

Reply:
https://github.com/ppy/osu/pull/32844/commits/3b2e8281b4e3f2ecdf5d6821427973d9004bb01d aside, looks good.